project/SparkBuild.scala

```diff
@@ -331,7 +331,13 @@ object SparkBuild extends PomBuild {
     javaOptions ++= {
       val versionParts = System.getProperty("java.version").split("[+.\\-]+", 3)
       var major = versionParts(0).toInt
-      if (major >= 16) Seq("--add-modules=jdk.incubator.vector,jdk.incubator.foreign", "-Dforeign.restricted=warn") else Seq.empty
+      if (major >= 21) {
```

Comment:
Maven doesn't need this options?

Reply:
This is a removal of `jdk.incubator.foreign`. This is the only place in our source code.

Reply:
Let me re-check~

Reply:
Yes, #32253 added this to SBT only.
```
$ git grep jdk.incubator.foreign
project/SparkBuild.scala:        Seq("--add-modules=jdk.incubator.vector,jdk.incubator.foreign", "-Dforeign.restricted=warn")
```